### PR TITLE
Add new CachedProjectEvaluation capability to default capabilities

### PR DESCRIPTION
**Pull request by arkalyanms**
Depends on https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/313873.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7066)

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProjectTypeCapabilities.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ internal class ProjectTypeCapabilities
                                       ProjectCapability.ProjectConfigurationsDeclaredDimensions + "; " +
                                       ProjectCapability.LanguageService + "; " +
                                       ProjectCapability.DotNet + "; " +
-                                      ProjectCapability.CachedProjectEvaluation;
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProjectTypeCapabilities.cs]: Are Shared Projects getting the same capability?

Comment (arkalyanms) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProjectTypeCapabilities.cs]: Yes, that's the plan. Just adding more testing to bring shared projects into the fold.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProjectTypeCapabilities.cs]: Have you verified F# projects work under this mode (you've opt'd them in)? They have a different way of displaying the tree so might be affected by this change.

Comment (arkalyanms) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProjectTypeCapabilities.cs]: Not yet, F# is in the test matrix for the cache service checkin acceptance. So far only been using CPS, project-system, orchardcore and some synthetic test solutions of varying size for the tests.

How are F# different? So I know what to test for.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProjectTypeCapabilities.cs]: Then specifically only opt in C# and VB: https://github.com/dotnet/project-system/blob/be0048c2e264608488391dee1a23921d47cb25fe/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProjectTypeCapabilities.cs#L46-L57.

F# opt's into SortByDisplayOrder which requires evaluation to provide items in the order they appear in the project file.

**Review by drewnoakes — APPROVED** [review]
@arkalyanms explained that the feature this capability lights up is behind a feature flag, at least initially. Adding this now unblocks other work.

**Comment by drewnoakes** [comment]
> error CS0117: 'ProjectCapabilities' does not contain a definition for 'UseProjectEvaluationCache'

Looks like the CPS package needs updating here.

**Pull request merged**